components/browser/menu/src/test/java/mozilla/components/browser/menu/view/DynamicWidthRecyclerViewTest.kt

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu.view
+
+import android.util.AttributeSet
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import mozilla.components.browser.menu.R
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers.anyInt
+import org.mockito.Mockito.never
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.robolectric.Robolectric
+import org.robolectric.annotation.Config
+
+@RunWith(AndroidJUnit4::class)
+class DynamicWidthRecyclerViewTest {
```

Comment:
Can we add a test here that this would work with `mozac_browser_menu_width` as well?

Reply:
Sorry but I'm not really sure what to test in this regards.
In the tests from line 33 and 45 I am testing the cases in which we will not try to calculate and set a "dynamic width". In which cases I verify that just the default behavior will be executed, onMeasure will use it's parameters - layout_width and layout_height. I added a small comment in this regard in DynamicWidthRecyclerView.
If you have other scenarios in mind I could use some pointers.

Reply:
@Mugurell I meant more specifically a case where you don't define `minWidth` and `maxWidth`? :)

Reply:
@Mugurell I meant more specifically a case where you don't define minWidth and maxWidth? :)

Reply:
Thanks!
I added 3 new tests:
- If minWidth and maxWidth are not provided view should use layout_width
- If only minWidth is provided view should use layout_width
- If only maxWidth is provided view should use layout_width